src/m365/aad/commands/app/app-get.ts

```diff
@@ -90,11 +89,74 @@ class AadAppGetCommand extends GraphCommand {
       });
   }
 
+  private getAppInfo(appObjectId: string): Promise<AppInfo> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/myorganization/applications/${appObjectId}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request.get<AppInfo>(requestOptions);
+  }
+
+  private saveAppInfo(args: CommandArgs, appInfo: AppInfo, logger: Logger): Promise<AppInfo> {
+    if (!args.options.save) {
+      return Promise.resolve(appInfo);
+    }
+
+    const filePath: string = '.m365rc.json';
+
+    if (this.verbose) {
+      logger.logToStderr(`Saving Azure AD app registration information to the ${filePath} file...`);
+    }
+
+    let m365rc: M365RcJson = {};
+    if (fs.existsSync(filePath)) {
+      if (this.debug) {
+        logger.logToStderr(`Reading existing ${filePath}...`);
+      }
+
+      try {
+        const fileContents: string = fs.readFileSync(filePath, 'utf8');
+        if (fileContents) {
+          m365rc = JSON.parse(fileContents);
+        }
+      }
+      catch (e) {
+        logger.logToStderr(`Error reading ${filePath}: ${e}. Please add app info to ${filePath} manually.`);
+        return Promise.resolve(appInfo);
+      }
+    }
+
+    if (!m365rc.apps) {
+      m365rc.apps = [];
+    }
+
+    if (m365rc.apps.every(a => a.appId !== appInfo.appId)) {
```

Comment:
Wouldn't it be more efficient to check if there's at least one element matching the appId rather than checking if every element is different? In the first case, we'd stop after the first match. In the latter, we keep continue checking other elements, which doesn't add any value.

Reply:
Absolutely right!